src/web/components/ParticipantManagement/AddParticipantDialog.tsx

```diff
@@ -0,0 +1,239 @@
+import Fuse from 'fuse.js';
+import { ChangeEvent, useEffect, useMemo, useState } from 'react';
+import { FormProvider, useForm } from 'react-hook-form';
+
+import { ApiRoleDTO } from '../../../api/entities/ApiRole';
+import { ParticipantTypeDTO } from '../../../api/entities/ParticipantType';
+import { UserRole } from '../../../api/entities/User';
+import { GetRecommendedRolesById, SiteDTO } from '../../../api/services/adminServiceHelpers';
+import { AddParticipantForm } from '../../services/participant';
+import { useSiteList } from '../../services/site';
+import { sortApiRoles } from '../../utils/apiRoles';
+import { Dialog } from '../Core/Dialog';
+import { CheckboxInput } from '../Input/CheckboxInput';
+import { Input } from '../Input/Input';
+import { RadioInput } from '../Input/RadioInput';
+import { SelectInput } from '../Input/SelectInput';
+import { TextInput } from '../Input/TextInput';
+import { SearchBarContainer, SearchBarInput, SearchBarResults } from '../Search/SearchBar';
+import { HighlightedResult } from './ParticipantApprovalForm';
+
+import './AddParticipantDialog.scss';
+
+type AddParticipantDialogProps = {
+  triggerButton: JSX.Element;
+  onAddParticipant: (form: AddParticipantForm) => Promise<void>;
+  apiRoles: ApiRoleDTO[];
+  participantTypes: ParticipantTypeDTO[];
+};
+
+function AddParticipantDialog({
+  triggerButton,
+  onAddParticipant,
+  apiRoles,
+  participantTypes,
+}: AddParticipantDialogProps) {
+  const { sites } = useSiteList();
+  const fuse = useMemo(
+    () =>
+      sites
+        ? new Fuse(sites!, { keys: ['name'], includeMatches: true, findAllMatches: true })
+        : null,
+    [sites]
+  );
+  const [open, setOpen] = useState(false);
+  const [siteSearchResults, setSiteSearchResults] = useState<Fuse.FuseResult<SiteDTO>[]>();
+  const [searchText, setSearchText] = useState('');
+  const [selectedSite, setSelectedSite] = useState<SiteDTO>();
+
+  const formMethods = useForm<AddParticipantForm>();
+  const { register, setValue, watch, handleSubmit, reset } = formMethods;
+
+  useEffect(() => {
+    setSiteSearchResults(fuse?.search(searchText));
+  }, [fuse, searchText]);
+
+  useEffect(() => {
+    const subscription = watch((value, { name }) => {
+      if (name === 'participantTypes') {
+        const types = value.participantTypes;
+        const roles = GetRecommendedRolesById(types as number[]);
+        setValue('apiRoles', roles);
+      }
+    });
+    return () => subscription.unsubscribe();
+  }, [watch, setValue, open]);
+
+  useEffect(() => {
+    if (!open) {
+      reset();
```

Comment:
I don't know what the fix is, but this doesn't seem to reset the SiteId search. Maybe can be fixed in a different ticket.

Reply:
I added this to https://atlassian.thetradedesk.com/jira/browse/UID2-2855.